### ISSUE TITLE
Discussion: VisualStateManager should provide more feedback when not placed in the right place

### ISSUE THREAD

**Issue by michael-hawker** [microsoft/microsoft-ui-xaml#5576]
# Discussion: Discussion: VisualStateManager should provide more feedback when not placed in the right place

Was trying to debug why our re-template of a control wasn't working, and only realized later after spending a couple of hours debugging and digging into the VSM states that the VSM wasn't defined in the root Grid of the template (the most likely cause).

The fact the VSM needs to be at the top/bottom of the top-level element of the control, isn't well documented, confusing, and easy to miss. If a VSM is initialized as part of a place where it's not going to work well, it should provide some warning or error to the app developer.

## Related Links
- #5575 
- #3289

**Comment by chingucoding** [comment]
This would make WinUI a lot easier to get into since this detail (give or take) can easily overlooked (also happened too often to me) and can be frustrating, especially since the VisualStateManager seems like it should be able to tell developers that something is off.

**Comment by StephenLPeters** [comment]
@RealTommyKlein is this something that we could enforce in the xaml compiler?

@chrisglein  FYI for the VSM aspect, raising an error here seems appropriate.

**Comment by chrisglein** [comment]
Agreed. Since it's not going to work elsewhere generating an error would be much more helpful.

**Comment by RealTommyKlein** [comment]
For the VisualStateManager case, that sounds reasonable - the Xaml compiler should be able to verify that named elements referenced by Setters really do exist.  I think it doesn't currently verify Setters today because the Xaml compiler evaluates each `.xaml` file independently, and there are more general cases for Setters where that isn't sufficient (e.g. Setters in a Style in a ResourceDictionary that is in its own separate `.xaml` file or `App.xaml` - then the Xaml compiler would need to be able to look up that Style when it's used in other files and evaluate it in the context of that file).
